clients/iota-go/iotajsonrpc/balance.go

```diff
@@ -32,6 +32,16 @@ func (t CoinType) String() string {
 	return string(t)
 }
 
+func (t *CoinType) MarshalBCS(e *bcs.Encoder) error {
+	rt, err := iotago.NewResourceType(t.String())
+	if err != nil {
+		return err
+	}
+	// use ShortString to save space
+	e.WriteString(rt.ShortString())
```

Comment:
In the beginning of our first Alpha we had an issue that we stored CoinTypes as ShortString but compared them with FullName Type strings, which caused several problems in the accounting contract. I'm not sure what the actual solution was. If we do a length agnostic check, or if we always use short/full length strings.

Just want to point out that we should validate this carefully, so we don't sometimes store full and short length types, or compare them incorrectly.

Reply:
Yes, that's why I defined other related types as `struct {s string}` instead of just `string`. For now let's keep it like this (if the tests pass) and when we get time to refactor properly let's unify the types and make sure this bug cannot happen.